nlmod/mdims/mgrid.py

```diff
@@ -195,15 +223,28 @@ def refine(
         model_ws = ds.model_ws
     g = Gridgen(dis, model_ws=model_ws, exe_name=exe_name)
 
+    ds_has_rotation = "angrot" in ds.attrs and ds.attrs["angrot"] != 0.0
+    if model_coordinates:
+        if not ds_has_rotation:
+            raise (Exception("The supplied shapes need to be in realworld coordinates"))
+    elif ds_has_rotation:
+        affine_matrix = get_affine_world_to_mod(ds).to_shapely()
+
     if refinement_features is not None:
         for refinement_feature in refinement_features:
             if len(refinement_feature) == 3:
                 # the feature is a file or a list of geometries
                 fname, geom_type, level = refinement_feature
+                if not model_coordinates and ds_has_rotation:
+                    raise (
+                        Exception("Converting files to model coordinates not supported")
```

Comment:
NotImplementedError instead of Exception?

Reply:
Fine. I never know which Exception I should pick, and therefore just take the most basic one.